client/web/src/components/series/label/SeriesLabelFrom.tsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react'
+import { useTranslation } from 'react-i18next'
+import { useDispatch } from 'react-redux'
+import { Formik } from 'formik'
+import {
+  TextField,
+  Button
+} from '@material-ui/core'
+
+import { actionTypes, SeriesService } from '../../../features/series'
+import { ILabel } from '../../../models/ILabel'
+import { ISeries } from '../../../models/series'
+
+type SeriesLabelFromProps = { 
+  series: ISeries, 
+  onCreate?: () => void ,
+  onCancel?: () => void
+}
+
+export const SeriesLabelFrom: React.FC<SeriesLabelFromProps> = (props: SeriesLabelFromProps) => {
+  const { t } = useTranslation()
+  const dispatch = useDispatch()
+
+  const onCancel = () => {
+    if(props.onCancel !== undefined) props.onCancel();
+  }
+
+  const validate = (values: ILabel) => {
+    let errors = {};
+    if (values.labelKeyId === '') {
+      errors = {"labelKeyId": 'Required'};
+    }
+    if (values.assignedValue === '') {
+      errors = {"assignedValue": 'Required'};
+    }
+    return errors;
+  };
+
+  const defaultValues = {
+    id: '',
+    userId: '',
+    labelKeyId: '',
+    labelTypeId: '',
+    labelValueId: '',
+    isPublic: '',
+    isApproved: '',
+    assignedValue: '',
+  }
+
+  return (
+      <Formik
+        enableReinitialize
+        initialValues = {defaultValues}
+        validate = {validate}
+        onSubmit={(values: ILabel) => {
+          if(props.series == null)
+          {
+            throw Error("No series selected")
+          }
+          if(props.series.labels == null)
+          {
+            props.series.labels = new Array<ILabel>();
+          }
+          props.series.labels.push(values);
+          SeriesService.update(props.series.id, props.series).then(()=> {dispatch({ type: actionTypes.FETCH_ALL_SERIES })});
```

Comment:
Following previous comment, you can change with: 

```typescript
dispatch({type: UPDATE_SERIES, series: props.series})
```
But You should assign props.series to one variable, to be sure there is no side effect

Reply:
Done